### ISSUE TITLE
[MNG-7897] Support ${project.rootDirectory} in file profile activation

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#9020]
**[Slawomir Jaranowski](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=slawekjaranowski)** opened **[MNG-7897](https://issues.apache.org/jira/browse/MNG-7897?redirect=false)** and commented

In current code we have support for short properties: `rootDirectory` to be consistence we should use `project.rootDirectory`

Property `project.rootDirectory` was introduced in MNG-7038, as we in alpha release we can change it.

---

**Issue Links:**
- [MNG-7895](https://issues.apache.org/jira/browse/MNG-7895) Support ${project.basedir} in file profile activation

**Remote Links:**
- [GitHub Pull Request #354
](https://github.com/apache/maven-integration-testing/pull/354)
- [GitHub Pull Request #1687
](https://github.com/apache/maven/pull/1687)

**Backported to:** [4.0.0-beta-4](https://github.com/apache/maven/milestone/11?closed=1)

**Comment by jira-importer** [comment]
**[Guillaume Nodet](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=gnt)** commented

[sjaranowski](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=sjaranowski) is that just about changing the following line:
https://github.com/apache/maven/blob/d075fe7e85d26d86ad25a963274cd9c5b910c56a/maven-api-impl/src/main/java/org/apache/maven/internal/impl/model/ProfileActivationFilePathInterpolator.java#L84 ?

There's also
https://github.com/apache/maven/blob/d075fe7e85d26d86ad25a963274cd9c5b910c56a/maven-api-impl/src/main/java/org/apache/maven/internal/impl/model/DefaultModelInterpolator.java#L222-L229

**Comment by jira-importer** [comment]
**[Jörg Hohwiller](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=hohwille)** commented

I found this feature here:

https://maven.apache.org/ref/4.0.0-rc-1/maven-model-builder/#model-interpolation

So I tested it with maven 4.0.0-rc-1 and it is not working:

```java
[WARNING] 'profiles.profile[maven3].activation.file.missing' Failed to interpolate profile activation property ${project.rootDirectory}/pom.xml: 
```

Does not make any difference if I use `{}${rootDirectory{`}} instead.

Will you consider to reopen or do I need to create a new bug issue?

BTW: I am trying to find a way for a profile activation based on the fact that I am on maven3 and not already on maven4. Then I could already move hacks like `resolve-parent-version-maven-plugin` or `flatten-maven-plugin` to such profile and prepare that my build works both with maven3 and maven4.
